src/backend/functions/projects-new.ts

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import middy from '@middy/core';
+import jsonBodyParser from '@middy/http-json-body-parser';
+import httpErrorHandler from '@middy/http-error-handler';
+import validator from '@middy/validator';
+import { Handler } from 'aws-lambda';
+import { PrismaClient, WBS_Element_Status } from '@prisma/client';
+import { buildClientFailureResponse, buildNotFoundResponse, buildSuccessResponse } from 'utils';
+
+const prisma = new PrismaClient();
+
+// gets highest current project number
+const getHighestProjectNumber = async (carNumber: number) => {
+  const maxProjectNumber = await prisma.wBS_Element.aggregate({
+    where: {
+      carNumber
+    },
+    max: {
+      projectNumber: true,
+    },
+  });
+
+  return maxProjectNumber.max.projectNumber;
+}
+
+// gets the associated change request for creating a project
+const getChangeRequestReviewState = async (crId: number) => {
+  const cr = await prisma.change_Request.findUnique({
+    where: {
+      crId
+    }
+  });
+
+  // returns null if the change request doesn't exist
+  // if it exists, return a boolean describing if the change request was reviewed
+  return cr ? cr.dateReviewed !== null : cr;
+}
+
+export const baseHandler: Handler = async ({ body }, _context) => {
+  // check if the change request exists
+  const crReviewed = await getChangeRequestReviewState(body.crId);
+  if (crReviewed === null) {
+    return buildNotFoundResponse('change request', `CR #${body.crId}`);
+  }
+
+  // check if the change request for the project was reviewed
+  if (!crReviewed) {
+    return buildClientFailureResponse('Cannot implement an unreview change request');
+  }
+
+  // create the wbs element for the project and document the implemented change request
+  const maxProjectNumber = await getHighestProjectNumber(body.carNumber);
+  const createdProject = await prisma.wBS_Element.create({
+    data: {
+      carNumber: body.carNumber,
+      projectNumber: maxProjectNumber + 1,
+      workPackageNumber: 0,
+      name: body.name,
+      status: WBS_Element_Status.INACTIVE,
```

Comment:
When I remove defining the status of the WBS Element, VS Code red underlines `data` and states the following:
```
(property) data: (Prisma.Without<Prisma.WBS_ElementCreateInput, Prisma.WBS_ElementUncheckedCreateInput> & Prisma.WBS_ElementUncheckedCreateInput) | (Prisma.Without<...> & Prisma.WBS_ElementCreateInput)
The data needed to create a WBS_Element.

Type '{ carNumber: any; projectNumber: number; workPackageNumber: number; name: any; project: { create: { summary: any; }; }; changes: { create: { changeRequestId: any; implementerId: any; detail: string; }; }; }' is not assignable to type '(Without<WBS_ElementCreateInput, WBS_ElementUncheckedCreateInput> & WBS_ElementUncheckedCreateInput) | (Without<...> & WBS_ElementCreateInput)'.
  Type '{ carNumber: any; projectNumber: number; workPackageNumber: number; name: any; project: { create: { summary: any; }; }; changes: { create: { changeRequestId: any; implementerId: any; detail: string; }; }; }' is not assignable to type 'Without<WBS_ElementUncheckedCreateInput, WBS_ElementCreateInput> & WBS_ElementCreateInput'.
    Property 'status' is missing in type '{ carNumber: any; projectNumber: number; workPackageNumber: number; name: any; project: { create: { summary: any; }; }; changes: { create: { changeRequestId: any; implementerId: any; detail: string; }; }; }' but required in type 'WBS_ElementCreateInput'.ts(2322)
index.d.ts(12757, 5): 'status' is declared here.
index.d.ts(8714, 5): The expected type comes from property 'data' which is declared here on type '{ select?: WBS_ElementSelect | null | undefined; include?: WBS_ElementInclude | null | undefined; data: (Without<WBS_ElementCreateInput, WBS_ElementUncheckedCreateInput> & WBS_ElementUncheckedCreateInput) | (Without<...> & WBS_ElementCreateInput); }'
```
For whatever reason, it wants me to define what `status` is. I think it might be that VS Code isn't updated with the types or something like that. I've already run `npm i` again to rebuild the types I have locally and the issue is still there. Do you have any ideas on why I'm bugged for excluding `status`?

Reply:
Weird! We can just leave it then